lib/test-rule.js

```diff
@@ -8,8 +8,18 @@ class Reporter {
     this.report = this.report.bind(this);
   }
 
-  report(id, message) {
-    this.messages.push({ id, message });
+  report(id, message, path) {
+
+    if (path) {
+      id = id.concat('#', stringifyPath(path));
```

Comment:
We want the ID and path to be formatted like this only when we log the results to the console (cf. https://github.com/bpmn-io/bpmnlint/blob/master/bin/bpmnlint.js#L109). The structure should be `{ id, message, path }`, otherwise we'd always have to parse the ID and path from the string to get them back.